libs/domain/cart/add/src/add.component.ts

```diff
@@ -137,47 +62,49 @@ export class CartAddComponent extends ProductComponentMixin<CartAddOptions>() {
     </oryx-button>`;
   }
 
-  /**
-   * Returns 0 when the max available quantity is smaller than the min order quantity.
-   */
-  protected calculateMin(
-    product: Product | undefined,
-    entries: CartEntry[]
-  ): number {
-    const maxAvailable = this.calculateMax(product, entries);
-    const minOrderQuantity = 1;
-    return maxAvailable !== undefined && maxAvailable < minOrderQuantity
-      ? 0
-      : minOrderQuantity;
-  }
-
-  /**
-   * When the product has a maximum quantity defined, we take the cumulated quantity from
-   * the active cart, and subtract this number from the available quantity.
-   */
-  protected calculateMax(
-    product: Product | undefined,
-    entries: CartEntry[]
-  ): number | undefined {
-    const cumulatedEntryCount = entries
-      .filter((entry) => entry.sku === product?.sku)
-      .map((entry) => entry.quantity)
-      .reduce((a: number, b) => a + b, 0);
-    return product?.availability?.quantity
-      ? product?.availability?.quantity - cumulatedEntryCount
-      : undefined;
-  }
+  protected min = computed(() => {
+    const value = 1;
+    if (this.$product()) {
+      const el = this.shadowRoot
+        ?.querySelector<QuantityInputComponent>('oryx-cart-quantity-input')
+        ?.reset();
+    }
+    return value;
+  });
+
+  protected max = computed(() => {
+    const qty = this.$product()?.availability?.quantity;
+    return qty
+      ? qty -
+          this.$entries()
+            .filter((entry) => entry.sku === this.$product()?.sku)
+            .map((entry) => entry.quantity)
+            .reduce((a: number, b) => a + b, 0)
+      : 0;
+  });
 
   protected onUpdate(e: CustomEvent<QuantityEventDetail>): void {
-    this.isQuantityInvalid$.next(!!e.detail.isInvalid);
+    this.isInvalid = !!e.detail.isInvalid;
   }
 
   protected onSubmit(e: Event | CustomEvent<QuantityEventDetail>): void {
     e.preventDefault();
-    this.triggerEntry$.next(
+    const sku = this.$product()?.sku;
+    if (!sku) return;
+
+    const quantity =
       (e as CustomEvent).detail?.quantity ??
-        this.quantityInputRef.value?.value ??
-        1
-    );
+      this.shadowRoot?.querySelector<QuantityInputComponent>(
+        'oryx-cart-quantity-input'
+      )?.value ??
+      this.min();
+
+    this.cartService
+      .addEntry({ sku, quantity })
+      .pipe(take(1))
+      .subscribe(() => {
```

Comment:
let's either add support for an error case (to not stuck in "in progress" state).